GVFS/GVFS.Platform.Windows/DiskLayoutUpgrades/DiskLayout12to13Upgrade_FolderPlaceholder.cs

```diff
@@ -81,30 +81,32 @@ private static IEnumerable<string> GetFolderPlaceholdersFromDisk(ITracer tracer,
                 {
                     if (!directory.EndsWith(Path.DirectorySeparatorChar + GVFSConstants.DotGit.Root))
                     {
-                        OnDiskFileState fileState = OnDiskFileState.Full;
-                        HResult result = Utils.GetOnDiskFileState(directory, out fileState);
-                        if (result == HResult.Ok)
+                        if (Directory.Exists(directory))

```

Comment:
I might have missed this in the earlier conversation; why was this `if (Directory.Exists(directory))` check added?

Reply:
We discussed this in one of our meetings with @cgallred.

Previously Utils.`GetOnDiskFileState `returned a result.  Now it has been turned into a bool.  
The only case we checked the HResult for previously was when `result != HResult.FileNotFound`.  In that instance we would not throw an exception.  We can simulate the same check by doing Directory.Exists upfront.

Please let me know if that makes sense or if I missed something!

Reply:
> The only case we checked the HResult for previously was when `result != HResult.FileNotFound`

I thought in our discussion with @cgallred we concluded that this check was only in the code to work around a  bug in RS4 where `TryGetOnDiskFileState` would return `HResult.FileNotFound` for tombstones and that moving forward VFS4G wasn't going to try to work around the Windows bug here.

> We can simulate the same check by doing Directory.Exists upfront.

If the directory is returned by `fileSystem.EnumerateDirectories(path)` would `Directory.Exists(directory)` return false?

If the `if (Directory.Exists(directory))` is adding value then I think we should include a comment explaining what the check is for.  If the `if (Directory.Exists(directory))` does not functionally change what we're doing here then it should be removed to reduce the complexity of this method.

Reply:
Yes, the only reason we were checking `HResult.FileNotFound` before was as a work around so I don't think we need this check unless we are continuing to work around that issue on a few machines that might continue to have the bug.

Reply:
@wilbaker @kewillford, check removed